tekton_configs/pytest-task.yaml

```diff
@@ -0,0 +1,110 @@
+apiVersion: tekton.dev/v1alpha1
+kind: Task
+metadata:
+  name: pytest
+  namespace: gabrielwen
+  annotations:
+    sidecar.istio.io/inject: "false"
+spec:
+  inputs:
+    params:
+    - name: test-worker-img
+      type: string
+      description: Test worker image.
+      default: gcr.io/kubeflow-ci/test-worker:latest
+    - name: src-dir
+      type: string
+      description: The path for source codes.
+      default: /mnt/testdir/src
+    - name: repos
+      type: string
+      description:
+        Repositories we need to checkout.
+    - name: python-path
+      type: string
+      description:
+        PYTHONPATH env.
+    - name: gcp-sa-dir
+      type: string
+      description:
+        GOOGLE_APPLICATION_CREDENTIALS path.
+      default: /secrets/sa
+    - name: get-cred-cmd
+      type: string
+      description:
+        Command used to run get-credentials.
+      default: python3
+    - name: get-cred-args-array
+      type: array
+      description:
+        Args passed into get-credentials cmd.
+    - name: pytest-args-array
+      type: array
+      description:
+        Args passed into pytest.
+    - name: pytest-wdr
+      type: string
+      description:
+        WorkingDir of pytest.
+    - name: test_job_type
+      type: string
+      description:
+        Job type of the test, from prow.
+      default: manual
+    - name: test_hostname
+      type: string
+      description:
+        host name of the test, from prow.
+      default: manual-testing
+  steps:
+  - name: git-checkout
```

Comment:
Why do you have a git checkout step? Why not just use Tekton git resources and let Tekton checkout the source for you?

Reply:
same as above, git resource is not able to change revision at runtime.